src/graphnet/models/gnn/dynedge_kaggle_tito.py

```diff
@@ -0,0 +1,253 @@
+                    pooling_scheme in GLOBAL_POOLINGS
+                ), f"Global pooling scheme {pooling_scheme} not supported."
+        else:
+            assert global_pooling_schemes is None
+
+        self._global_pooling_schemes = global_pooling_schemes
+
+        assert self._global_pooling_schemes, (
+            "No global pooling schemes were request, so cannot add global"
+            " variables after pooling."
+        )
+
+        # Base class constructor
+        super().__init__(nb_inputs, self._readout_layer_sizes[-1])
+
+        # Remaining member variables()
+        self._activation = torch.nn.LeakyReLU()
+        self._nb_inputs = nb_inputs
+        self._nb_global_variables = 5 + nb_inputs
+        self._features_subset = features_subset
+        self._construct_layers()
+
+    def _construct_layers(self) -> None:
+        """Construct layers (torch.nn.Modules)."""
+        # Convolutional operations
+        nb_input_features = self._nb_inputs
+
+        self._conv_layers = torch.nn.ModuleList()
+        nb_latent_features = nb_input_features
+        for sizes in self._dynedge_layer_sizes:
+            conv_layer = DynTrans(
+                [nb_latent_features] + list(sizes),
+                aggr="max",
+                nb_neighbors=6,
+                features_subset=self._features_subset,
+            )
+            self._conv_layers.append(conv_layer)
+            nb_latent_features = sizes[-1]
+
+        # Post-processing operations
+        nb_latent_features = self._dynedge_layer_sizes[-1][-1]
+
+        post_processing_layers = []
+        layer_sizes = [nb_latent_features] + list(
+            self._post_processing_layer_sizes
+        )
+        for nb_in, nb_out in zip(layer_sizes[:-1], layer_sizes[1:]):
+            post_processing_layers.append(torch.nn.Linear(nb_in, nb_out))
+            post_processing_layers.append(self._activation)
+        last_posting_layer_output_dim = nb_out
+
+        self._post_processing = torch.nn.Sequential(*post_processing_layers)
+
+        # Read-out operations
+        nb_poolings = (
+            len(self._global_pooling_schemes)
+            if self._global_pooling_schemes
+            else 1
+        )
+        nb_latent_features = last_posting_layer_output_dim * nb_poolings
+        nb_latent_features += self._nb_global_variables
+
+        readout_layers = []
+        layer_sizes = [nb_latent_features] + list(self._readout_layer_sizes)
+        for nb_in, nb_out in zip(layer_sizes[:-1], layer_sizes[1:]):
+            readout_layers.append(torch.nn.Linear(nb_in, nb_out))
+            readout_layers.append(self._activation)
+
+        self._readout = torch.nn.Sequential(*readout_layers)
+
+        # Transformer layer(s)
+        encoder_layer = TransformerEncoderLayer(
+            d_model=last_posting_layer_output_dim,
+            nhead=8,
+            batch_first=True,
+            norm_first=False,
+        )
+        self._transformer_encoder = TransformerEncoder(
```

Comment:
Because `num_layers = 0`, `self._transformer_encoder` is actually set to 
```
<bound method Module.state_dict of TransformerEncoder(
  (layers): ModuleList()
)>
```
which looks like an empty `ModuleList`, so no learnable weights. Is this intentional? It appears to me that this is redundant code. Also, the diagram shown in the Kaggle solution doesn't appear to have a transformer readout-layer, which this would be, so I think this was his way of turning it off. :-)

Reply:
That's a nice observation, you're right. I will delete this empty layer